admiral/pkg/clusters/serviceentry.go

```diff
@@ -175,33 +176,82 @@ func createServiceEntryForNewServiceOrPod(env string, sourceIdentity string, rem
 				}
 			}
 
-			//add virtual service for routing locally in within the cluster
-			//virtualServiceName := getIstioResourceName(cname, "-default-vs")
-			//
-			//oldVirtualService := rc.IstioConfigStore.Get(istioModel.VirtualService.Type, virtualServiceName, remoteRegistry.config.SyncNamespace)
-			//
-			//virtualService := makeVirtualService(serviceEntry.Hosts[0], localFqdn, meshPorts[common.Http])
-			//
-			//newVirtualService, err := createIstioConfig(istio.VirtualServiceProto, virtualService, virtualServiceName, remoteRegistry.config.SyncNamespace)
-			//
-			//if err == nil {
-			//	addUpdateIstioResource(rc, *newVirtualService, oldVirtualService, virtualServiceName, syncNamespace)
-			//} else {
-			//	log.Errorf(LogErrFormat, "Create", istioModel.VirtualService.Type, virtualServiceName, rc.ClusterID, err)
-			//}
+			for _, val := range dependents.Map() {
+				remoteRegistry.AdmiralCache.DependencyNamespaceCache.Put(val, serviceInstance.Namespace, localFqdn)
+			}
+
+			if common.GetWorkloadSidecarUpdate() == "enabled" {
+				for _, sidecarEgress := range remoteRegistry.AdmiralCache.DependencyNamespaceCache.Get(sourceIdentity) {
+					modifySidecarForLocalClusterCommunication(serviceInstance.Namespace, sidecarEgress.Namespace, sidecarEgress.FQDN, rc)
+				}
+			}
 		}
 	}
 	return serviceEntries
 }
 
+func modifySidecarForLocalClusterCommunication(sidecarNamespace string, dependencyNamespace string, localFqdn string, rc *RemoteController) {
+
+	//get existing sidecar from the cluster
+	sidecarConfig := rc.SidecarController
+
+	if sidecarConfig == nil {
+		return
+	}
+
+	sidecar, _ := sidecarConfig.IstioClient.NetworkingV1alpha3().Sidecars(sidecarNamespace).Get(common.GetWorkloadSidecarName(), v12.GetOptions{})
+
+	if sidecar == nil || sidecar.Spec.Egress == nil {
```

Comment:
Make sure `sidecar.Spec` always returns a value when `sidecar` is not nil.

Reply:
Yes, verified it as part of test - TestModifyNonExistingSidecarForLocalClusterCommunication

Spec is present when sidecar is not nil